src/main/scala/io/iohk/ethereum/mining/BlockGenerator.scala

```diff
@@ -37,11 +37,13 @@ class BlockGenerator(blockchainStorages: BlockchainStorages, blockchainConfig: B
         val blockTimestamp = blockTimestampProvider.getEpochSecond
         val header: BlockHeader = prepareHeader(blockNumber, ommers, beneficiary, parent, blockTimestamp)
 
-        val transactionsForBlock = transactions
-          .filter(_.tx.gasLimit < header.gasLimit)
-          //if we have 2 transactions from same address we want first one with lower nonce
-          .sortBy(_.tx.gasPrice)
-          .sortBy(_.tx.nonce)
+        val sortedTransactions = transactions.groupBy(_.senderAddress).values.toList.flatMap { txsFromSender =>
+          val ordered = txsFromSender.sortBy(_.tx.gasPrice).reverse.sortBy(_.tx.nonce)
+            .takeWhile(_.tx.gasLimit <= header.gasLimit)
```

Comment:
if we will not check it here all transactions after this will fail (from that account) and we will include less valid transactions in block
but this is mainly because here
https://github.com/input-output-hk/etc-client/pull/243/files#diff-6d3634062464c02321dc5a991254d727R49
we would prepare empty block if tx with gas limit set to > blockGasLimit would be first